Cleanup APIs

sgtcoolguy opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-27617

**Description:**
Now that we have nice npm scripts to see what's deprecated in our APIs but not removed (See #11366), it gave me a good list of things to look over. This PR is a work in progress to remove those long-deprecated APIs (or fix up our docs to update when they actually did get removed).

The commit messages detail the exact APIs I'm working with, but I'll try to summarize here:
- `Ti.UI.CoverFlowView` was removed in 6.0.0 (moved under iOS namespace)
- `Ti.UI.AdView` was removed in 7.0.0 (basically the create method was around and just logged a message)
- `Ti.UI.Android.LINKIFY_*` constants were removed here for 9.0.0 (use `Ti.UI.AUTOLINK_*` constants)
- `Ti.Stream` callbacks' `errorState` and `errorDescription` properties were removed here for 9.0.0 (use `code` and `error` properties)
- `Ti.Network.Socket.TCP` error callback `errorCode` property was removed here for 9.0.0 (use `code`)
- `Ti.UI.WebView` error event `errorCode` and `message` properties were removed here for 9.0.0 (use `code` and `error`)
- `Ti.UI.iPad.Popover` properties `height`, `width`, `leftNavButton` and `rightNavButton` were removed (the nav buttons were actually removed at time of deprecation in 3.4.2)
- `Ti.UI.Window` `android:*` events are removed here for 9.0.0 (replaced by equivalent without colon in 3.0.0)
- `Ti.Media.cameraAuthorizationStatus` was removed here for 9.0.0 (replaced by `Ti.Media.cameraAuthorization` in 5.2.0)
- `Ti.Media. AUDIO_SESSION_MODE_SOLO_AMBIENT` was actually removed in 6.0.0
- `Ti.Media. AUDIO_SPEAKER` was actually removed in 3.6.0
- `Ti.Media.recordinginput` and `Ti.Media.linechange` events were actually removed in 3.5.1
- `Ti.UI.View`'s `startLayout()`, `updateLayout()` and `finishLayout()` methods are removed here for 9.0.0 (deprecated since 3.0.0)
- `Ti.UI.iOS.BLEND_MODE_*` constants are removed here for 9.0.0 (replaced by equivalents in `Ti.UI` namespace in 7.3.0)

sgtcoolguy committed:
docs: mark ti.ui.coverflowview removed as of 6.0.0

sgtcoolguy committed:
docs: mark ti.ui.ios.adview removed as of 7.0.0

sgtcoolguy committed:
refactor(android): remove deprecated linkify constants

BREAKING CHANGE: Removes the LINKIFY_* constants living in Android namespace taht have been
deprecated since SDK 3.0.0

sgtcoolguy committed:
docs: mark linkify constants as removed as of sdk 9.0.0

sgtcoolguy committed:
refactor(android): remove deprecated properties on stream callbacks

This removes the errorState and errorDescription properties that have been deprecated since sdk
3.0.0

BREAKING CHANGE: Removes deprecated stream callback properties

sgtcoolguy committed:
refactor(ios): remove deprecated properties on stream callbacks

    This removes the errorState and errorDescription properties that have been deprecated since sdk
    3.0.0

    BREAKING CHANGE: Removes deprecated stream callback properties

sgtcoolguy committed:
docs: mark stream callback properties removed as of 9.0.0

Marks the errorState and errorDescription properties on stream callbacks removed in SDK 9.0.0 that have been deprecated since 3.0.0

sgtcoolguy committed:
refactor(android): remove deprecated property on tcp error callback

    This removes the errorCode property that has been deprecated since sdk
    3.1.0

    BREAKING CHANGE: Removes deprecated tcp error callback errorCode property

sgtcoolguy committed:
refactor(ios): remove deprecated property on tcp error callback

    This removes the errorCode property that has been deprecated since sdk
    3.1.0

    BREAKING CHANGE: Removes deprecated tcp error callback errorCode property

sgtcoolguy committed:
docs: mark tcp error callback property removed as of 9.0.0

Marks the errorCode property on tcp error callbacks removed in SDK 9.0.0 that have been deprecated since 3.1.0

sgtcoolguy committed:
refactor(android): remove deprecated properties on webview error event

    This removes the errorCode and message properties that have been deprecated since sdk
    3.1.0

    BREAKING CHANGE: Removes deprecated webview error event properties: errorCode and message

sgtcoolguy committed:
docs: mark webview error event properties removed as of 9.0.0

Marks the errorCode and message properties on webview error events removed in SDK 9.0.0 that have been deprecated since 3.1.0

sgtcoolguy committed:
refactor(ios): remove deprecated properties on ipad popover

    This removes the height, width, leftNavButton and rightNavButton properties that have been deprecated since sdk
    3.4.2

    BREAKING CHANGE: Removes deprecated ipad popover properties: height, width, leftNavButton and rightNavButton

sgtcoolguy committed:
docs: mark ipad popover properties removed as of 9.0.0

Marks the height, width, leftNavButton and rightNavButton properties on
popover removed in SDK 9.0.0 that have been deprecated since 3.4.2.
Note that some of them were effectively removed at time of deprecation.

build commented:
<!--
  0 failure: 
  1 warning:  There is no linke...
  3 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a></td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:thumbsup: Hey!, You deleted more code than you added. That's awesome!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 4412 tests are passing.
(There are 479 skipped tests not included in that total)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3300d49828a92463ce91f5b5daec024944f75a43
</p>

sgtcoolguy committed:
refactor(android): remove deprecated ti.ui.window android:* events

BREAKING CHANGES: removes deprecated Ti.UI.Window.android:* events

sgtcoolguy committed:
docs: mark android:* events as removed on ti.ui.window

sgtcoolguy committed:
docs: report various properties/events as removed with correct versions

sgtcoolguy committed:
refactor(ios): remove Ti.Media.cameraAuthorizationStatus

BREAKING CHANGES: removes deprecated Ti.Media.cameraAuthorizationStatus property

sgtcoolguy committed:
docs: mark Ti.Media.cameraAuthorizationStatus removed in 9.0.0

sgtcoolguy committed:
refactor(ios): remove deprecated view methods around layout lifecycle

BREAKING CHANGE: Removes the updateLayout(), startLayout() and finishLayout() methods on Ti.UI.View

sgtcoolguy committed:
docs: mark Ti.UI.View layout lifecycle methods removed in 9.0.0

sgtcoolguy committed:
refactor(ios): remove deprecated Ti.UI.iOS.BLEND_MODE constants

BREAKING CHANGE: Removes the BLEND_MODE_* constants on Ti.UI.iOS namespace (use constants on Ti.UI namespace)

sgtcoolguy committed:
docs: mark Ti.UI.iOS.BLEND_MODE_* constants removed in 9.0.0

sgtcoolguy commented:
Relates to #11362

sgtcoolguy committed:
Merge branch 'master' into cleanup-apis

ypbnv reviewed (approved):
Android changes look good to me!

vijaysingh-axway reviewed (approved):
CR passed for iOS.
